test/integration/sequelize.test.js

```diff
@@ -807,9 +807,28 @@ describe(Support.getTestDialectTeaser('Sequelize'), () => {
     }
   });
 
-  describe('databaseVersion', () => {
+  describe('fetchDatabaseVersion', () => {
     it('should database/dialect version', async function () {
-      const version = await this.sequelize.databaseVersion();
+      const version = await this.sequelize.fetchDatabaseVersion();
+      expect(typeof version).to.equal('string');
+      expect(version).to.be.ok;
+    });
+  });
+
+  describe('getDatabaseVersion', () => {
+    it('throws if no database version is set internally', () => {
+      expect(() => {
+        // ensures the version hasn't been loaded by another test yet
+        const sequelize = Support.createSequelizeInstance();
+        sequelize.getDatabaseVersion();
+      }).to.throw(
+        'The current database version is unknown, please call `sequelize.authenticate()` first to fetch it. Or manually configure it through options.',
+      );
+    });
+
+    it('should be able to get database version once loaded internally', async function () {
```

Comment:
I try to avoid the `should` wording because I find it unnecessarily verbose

```suggestion
    it('returns the database version if loaded', async function () {
```